frb_rust/src/handler.rs

```diff
@@ -241,13 +252,30 @@ impl<EH: ErrorHandler> Executor for ThreadPoolExecutor<EH> {
                         }
                     }
                     Err(error) => {
-                        eh2.handle_error(wrap_info2.port.unwrap(), Error::ResultError(error));
+                        eh2.handle_error(port2, Error::ResultError(error));
                     }
                 };
             });
 
             if let Err(error) = thread_result {
-                eh.handle_error(wrap_info.port.unwrap(), Error::Panic(error));
+                eh.handle_error(port.expect("(worker) eh"), Error::Panic(error));
+            }
+        };
+        #[cfg(not(target_family = "wasm"))]
+        THREAD_POOL.lock().execute(move || worker(port));
+
+        #[cfg(target_family = "wasm")]
+        WORKER_POOL.with(|pool| {
+            use wasm_bindgen::{JsCast, JsValue};
+            // MessagePort is not thread-safe (as it is only a pointer to a JS scope-local heap)
+            // but is transferrable, so we have to delegate to JS for marshalling.
+            // TODO(Desdaemon): Make a thread-safe adapter for MessagePort
```

Comment:
Agree. Seems that we are hardcoding too many low-level things in this single handler function...

Reply:
Yeah, this ties into the "ZeroCopyBuffer is not zerocopy on web" situation, since there is a need to know the parameters statically to accurately pass only transferables. A macro might be sufficient.

Reply:
> there is a need to know the parameters statically to accurately pass only transferables. A macro might be sufficient.

I see. Not sure about the details, but add my two cents: Or maybe a trait and some impls?